artichoke-backend/src/extn/core/env/env.rb

```diff
@@ -1,3 +1,119 @@
 # frozen_string_literal: true
 
-ENV = EnvClass.new
+class EnvClass
+  def assoc(name)
+    value = self[name]
+
+    value.nil? ? nil : [name, self[name]]
+  end
+
+  def clear
+    to_h.each do |var_name, var_value|
+      self[var_name] = nil unless var_value.nil?
+    end
+
+    to_h
+  end
+
+  def delete(name)
+    value = self[name]
+    return nil if value.nil?
+
+    self[name] = nil
+    yield name if block_given?
+
+    value
+  end
+
+  def delete_if
+    return to_enum(:delete_if) unless block_given?
+
+    each do |key, value|
```

Comment:
Are we missing a definition of `each`?

`clear` calls `to_h.each`. Should we be doing that here?

Reply:
I'm sorry, I'm not familiar with ruby (and didn't check it), it's my fault.